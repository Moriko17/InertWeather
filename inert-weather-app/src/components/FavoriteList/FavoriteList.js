import React from "react";
import "../../styles/styles.css";
import FavoriteItem from "../FavoriteItem/FavoriteItem";

class FavoriteList extends React.Component {
  componentDidMount() {
    this.props
      .fetchFavoritesList()
      .then(response => {
        let cities = [];
        response.data.forEach(city => {
          cities.push({
            cityName: city.cityName,
            id: cities.length + 1
          });
        });
        const newState = {
          cities: cities,
          lastUsedId: cities.length + 1
        };
        this.props.updateState(newState);
      })
      .catch(error => {
        console.log(error);
      });
  }
  render() {
    return (
      <div className="favorite-list">
        {this.props.favoriteData.cities.map(city => (
          <FavoriteItem
            key={city.id}
            city={city}
            deleteCity={this.props.deleteCity}
            deleteCityOnSrv={this.props.deleteCityOnSrv}
            fetchWeather={this.props.fetchWeather}
          />
        ))}
      </div>
    );
  }
}

export default FavoriteList;
