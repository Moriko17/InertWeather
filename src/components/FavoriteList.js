import React from "react";
import "../styles/styles.css";
import FavoriteItem from "./FavoriteItem";

class FavoriteList extends React.Component {
  render() {
    console.log(this.props);
    return (
      <div>
        {this.props.favoriteData.cities.map(city => (
          <FavoriteItem
            key={city.id}
            city={city}
            deleteCity={this.props.deleteCity}
            fetchWeather={this.props.fetchWeather}
          />
        ))}
      </div>
    );
  }
}

export default FavoriteList;
