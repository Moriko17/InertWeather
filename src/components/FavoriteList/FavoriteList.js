import React from "react";
import "../../styles/styles.css";
import FavoriteItem from "../FavoriteItem/FavoriteItem";

class FavoriteList extends React.Component {
  componentDidUpdate() {
    let dataToStorage = this.props.favoriteData.cities.filter(
      city => city.fetchedData
    );
    dataToStorage = dataToStorage.map(city => ({
      id: city.id,
      cityName: city.cityName
    }));
    localStorage.setItem("cities", JSON.stringify(dataToStorage));
    localStorage.setItem("lastUsedId", this.props.favoriteData.lastUsedId);
  }
  render() {
    return (
      <div className="favorite-list">
        {this.props.favoriteData.cities.map(city => (
          <FavoriteItem
            key={city.id}
            city={city}
            deleteCity={this.props.deleteCity}
            fetchWeather={this.props.fetchWeather}
          />
        ))}
      </div>
    );
  }
}

export default FavoriteList;
