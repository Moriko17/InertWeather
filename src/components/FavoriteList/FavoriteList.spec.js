import React from "react";
import renderer from "react-test-renderer";
import FavoriteList from "./FavoriteList";

describe("FavoriteList renders right", () => {
  const favoriteDataWithOneCity = {
    cities: [
      {
        cityName: "Marunouchi",
        erroeMessage: "",
        isLoading: false,
        fetchedData: {
          cityName: "Marunouchi",
          cloudState: "few clouds",
          coord: {
            lat: 35.68,
            lon: 139.76
          },
          humidity: 63,
          pressure: 1014,
          temp: 13.11,
          wind: 2.6
        },
        id: 60
      }
    ],
    lastUsedId: 60
  };

  const favoriteDataWithCities = {
    cities: [
      {
        cityName: "Marunouchi",
        erroeMessage: "",
        isLoading: false,
        fetchedData: {
          cityName: "Marunouchi",
          cloudState: "few clouds",
          coord: {
            lat: 35.68,
            lon: 139.76
          },
          humidity: 63,
          pressure: 1014,
          temp: 13.11,
          wind: 2.6
        },
        id: 60
      },
      {
        cityName: "Marunouchi",
        erroeMessage: "",
        isLoading: false,
        fetchedData: {
          cityName: "Marunouchi",
          cloudState: "few clouds",
          coord: {
            lat: 35.68,
            lon: 139.76
          },
          humidity: 63,
          pressure: 1014,
          temp: 13.11,
          wind: 2.6
        },
        id: 61
      },
      {
        cityName: "Marunouchi",
        erroeMessage: "",
        isLoading: false,
        fetchedData: {
          cityName: "Marunouchi",
          cloudState: "few clouds",
          coord: {
            lat: 35.68,
            lon: 139.76
          },
          humidity: 63,
          pressure: 1014,
          temp: 13.11,
          wind: 2.6
        },
        id: 62
      }
    ],
    lastUsedId: 62
  };

  const favoriteDataBlank = {
    cities: [],
    lastUsedId: 70
  };

  it("Renders correctly with one city", () => {
    const tree = renderer
      .create(
        <FavoriteList
          fetchWeather={() => {}}
          deleteCity={() => {}}
          favoriteData={favoriteDataWithOneCity}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renders correctly without cities", () => {
    const tree = renderer
      .create(
        <FavoriteList
          fetchWeather={() => {}}
          deleteCity={() => {}}
          favoriteData={favoriteDataBlank}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Renders correctly with many sities", () => {
    const tree = renderer
      .create(
        <FavoriteList
          fetchWeather={() => {}}
          deleteCity={() => {}}
          favoriteData={favoriteDataWithCities}
        />
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
