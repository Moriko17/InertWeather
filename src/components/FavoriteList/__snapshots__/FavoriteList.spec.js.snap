// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FavoriteList renders right Renders correctly with many sities 1`] = `
<div
  className="favorite-list"
>
  <div
    className="favorite-container"
  >
    <div
      className="favorite-bar"
    >
      <p>
        Marunouchi
      </p>
      <p>
        13.11
         째C
      </p>
      <button
        onClick={[Function]}
      >
        X
      </button>
    </div>
    <div
      className="error"
    >
      <p>
        Something goes wrong: 
      </p>
    </div>
  </div>
  <div
    className="favorite-container"
  >
    <div
      className="favorite-bar"
    >
      <p>
        Marunouchi
      </p>
      <p>
        13.11
         째C
      </p>
      <button
        onClick={[Function]}
      >
        X
      </button>
    </div>
    <div
      className="error"
    >
      <p>
        Something goes wrong: 
      </p>
    </div>
  </div>
  <div
    className="favorite-container"
  >
    <div
      className="favorite-bar"
    >
      <p>
        Marunouchi
      </p>
      <p>
        13.11
         째C
      </p>
      <button
        onClick={[Function]}
      >
        X
      </button>
    </div>
    <div
      className="error"
    >
      <p>
        Something goes wrong: 
      </p>
    </div>
  </div>
</div>
`;

exports[`FavoriteList renders right Renders correctly with one city 1`] = `
<div
  className="favorite-list"
>
  <div
    className="favorite-container"
  >
    <div
      className="favorite-bar"
    >
      <p>
        Marunouchi
      </p>
      <p>
        13.11
         째C
      </p>
      <button
        onClick={[Function]}
      >
        X
      </button>
    </div>
    <div
      className="error"
    >
      <p>
        Something goes wrong: 
      </p>
    </div>
  </div>
</div>
`;

exports[`FavoriteList renders right Renders correctly without cities 1`] = `
<div
  className="favorite-list"
/>
`;
